core/src/main/java/io/questdb/griffin/engine/groupby/MonthTimestampMicrosSampler.java

```diff
@@ -57,9 +57,9 @@ public long nextTimestamp(long timestamp) {
     }
 
     @Override
-    public long nextTimestamp(long timestamp, int numSteps) {
+    public long nextTimestamp(long timestamp, long numSteps) {
         try {
-            return addMonth(timestamp, Math.multiplyExact(numSteps, stepMonths));
+            return addMonth(timestamp, (int) Math.multiplyExact(numSteps, stepMonths));
```

Comment:
Since we're handling `ArithmeticException`, we should probably use `Math#toIntExact()` instead of the int cast here.

Reply:
fixed in https://github.com/questdb/questdb/pull/6517/commits/c9c81292ac6c7c9e4a5bd20c2a21603d51c03623